benchmarks/ekomark/benchmark/external/apfel_utils.py

```diff
@@ -0,0 +1,231 @@
+    apfel.SetSin2ThetaW(theory["SIN2TW"])
+
+    apfel.SetCKM(*[float(x) for x in theory.get("CKM").split()])
+
+    # TMCs
+    apfel.SetProtonMass(theory.get("MP"))
+    if theory.get("TMC"):
+        apfel.EnableTargetMassCorrections(True)
+
+    # Heavy Quark Masses
+    if theory.get("HQ") == "POLE":
+        apfel.SetPoleMasses(theory.get("mc"), theory.get("mb"), theory.get("mt"))
+    elif theory.get("HQ") == "MSBAR":
+        apfel.SetMSbarMasses(theory.get("mc"), theory.get("mb"), theory.get("mt"))
+        apfel.SetMassScaleReference(
+            theory.get("Qmc"), theory.get("Qmb"), theory.get("Qmt")
+        )
+    else:
+        raise RuntimeError("Error: Unrecognised HQMASS")
+
+    # Heavy Quark schemes
+    apfel.SetMassScheme(theory.get("FNS"))
+    apfel.EnableDampingFONLL(theory.get("DAMP"))
+    if theory.get("FNS") == "FFNS":
+        apfel.SetFFNS(theory.get("NfFF"))
+        apfel.SetMassScheme("FFNS%d" % theory.get("NfFF"))
+    else:
+        apfel.SetVFNS()
+
+    apfel.SetMaxFlavourAlpha(theory.get("MaxNfAs"))
+    apfel.SetMaxFlavourPDFs(theory.get("MaxNfPdf"))
+
+    # Scale ratios
+    apfel.SetRenFacRatio(theory.get("XIR") / theory.get("XIF"))
+    apfel.SetRenQRatio(theory.get("XIR"))
+    apfel.SetFacQRatio(theory.get("XIF"))
+    # Scale Variations
+    # consistent with Evolution (0) or DIS only (1)
+    # look at SetScaleVariationProcedure.f
+    apfel.SetScaleVariationProcedure(theory.get("EScaleVar"))
+
+    # Small-x resummation
+    apfel.SetSmallxResummation(theory.get("SxRes"), theory.get("SxOrd"))
+    apfel.SetMassMatchingScales(
+        theory.get("kcThr"), theory.get("kbThr"), theory.get("ktThr")
+    )
+
+    # Intrinsic charm
+    apfel.EnableIntrinsicCharm(theory.get("IC"))
+
+    # Not included in the map
+    #
+    # Truncated Epsilon
+    # APFEL::SetEpsilonTruncation(1E-1);
+    #
+    # Set maximum scale
+    # APFEL::SetQLimits(theory.Q0, theory.QM );
+    #
+    # if (theory.SIA)
+    # {
+    #   APFEL::SetPDFSet("kretzer");
+    #   APFEL::SetTimeLikeEvolution(true);
+    # }
+
+    # Set APFEL interpolation grid
+    #
+    # apfel.SetNumberOfGrids(3)
+    # apfel.SetGridParameters(1, 50, 3, 1e-5)
+    # apfel.SetGridParameters(2, 50, 3, 2e-1)
+    # apfel.SetGridParameters(3, 50, 3, 8e-1)
+
+    # set APFEL grid to ours
+    if platform.node() in ["FHe19b", "topolinia-arch"]:
+        apfel.SetNumberOfGrids(1)
+        # create a 'double *' using swig wrapper
+        yad_xgrid = operators["interpolation_xgrid"]
+        xgrid = apfel.new_doubles(len(yad_xgrid))
+
+        # fill the xgrid with
+        for j, x in enumerate(yad_xgrid):
+            apfel.doubles_setitem(xgrid, j, x)
+
+        yad_deg = operators["interpolation_polynomial_degree"]
+        # 1 = gridnumber
+        apfel.SetExternalGrid(1, len(yad_xgrid) - 1, yad_deg, xgrid)
+
+    # set DIS params
+    apfel.SetPDFSet(pdf)
+    # set Target
+
+    # init evolution
+    apfel.InitializeAPFEL()
+
+    return apfel
```

Comment:
This function it's more or less the same between `ekomark` and `yadmark`: should we lift at least part of it in banana? @felixhekhorn

Reply:
see https://github.com/N3PDF/banana/pull/13